src/clj/game/cards/ice.clj

```diff
@@ -1212,12 +1212,12 @@
                  end-the-run-if-tagged]})
 
 (defcard "Datapike"
-  {:subroutines [{:msg "force the Runner to pay 2 [Credits] if able"
-                  :async true
+  {:subroutines [{:async true
                   :effect (req (wait-for (pay state :runner (make-eid state eid) card :credit 2)
-                                         (when-let [payment-str (:msg async-result)]
-                                           (system-msg state :runner payment-str))
-                                         (effect-completed state side eid)))}
+                                         (if (:cost-paid async-result)
```

Comment:
Oh man, we should be using this everywhere instead of checking the msg, good call.

Reply:
I applied a similar update to 2 more functions, I hope they are ok